src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -138,11 +139,32 @@ private async Task<IAccount> GetSingleAccountForLoginHintAsync(string loginHint)
         {
             if (!string.IsNullOrEmpty(loginHint))
             {
-                IReadOnlyList<IAccount> accounts = (await CacheManager.GetAccountsAsync().ConfigureAwait(false))
+                IList<IAccount> accounts = (await CacheManager.GetAccountsAsync().ConfigureAwait(false))
                     .Where(a => !string.IsNullOrWhiteSpace(a.Username) &&
                            a.Username.Equals(loginHint, StringComparison.OrdinalIgnoreCase))
                     .ToList();
 
+                if (accounts.Count == 0 && DesktopOsHelper.IsWindows())

```

Comment:
we'd probably need to add `if brokerEnabled`

Reply:
Should I do this *instead of* the OS check?